src/nile/common.py

```diff
@@ -28,25 +28,36 @@
     # subject to change
     "0x041a78e741e5af2fec34b695679bc6891742439f7afb8484ecd7766661ad02bf"
 )
+DEFAULT_GATEWAYS = {
+    "localhost": "http://127.0.0.1:5050/",
+    "goerli2": "https://alpha4-2.starknet.io",
+    "integration": "https://external.integration.starknet.io",
+}
 
 
 def get_gateways():
     """Get the StarkNet node details."""
-    try:
+    if os.path.exists(NODE_FILENAME):
         with open(NODE_FILENAME, "r") as f:
             gateway = json.load(f)
-            return gateway
-
-    except FileNotFoundError:
-        with open(NODE_FILENAME, "w") as f:
-            networks = {
-                "localhost": "http://127.0.0.1:5050/",
-                "goerli2": "https://alpha4-2.starknet.io",
-                "integration": "https://external.integration.starknet.io",
-            }
-            f.write(json.dumps(networks, indent=2))
-
-            return networks
+            gateways = {**DEFAULT_GATEWAYS, **gateway}
+            return gateways
+    else:
+        return DEFAULT_GATEWAYS
+
+
+def create_node_json(network, gateway_url):
```

Comment:
```suggestion
def write_node_json(network, gateway_url):
```
it only creates if it doesn't exist